spec/system/consumer/split_checkout_spec.rb

```diff
@@ -85,4 +86,64 @@
       expect(page).to have_current_path("/checkout/payment")
     end
   end
+
+  describe 'login in as user' do
+    let(:user) { create(:user) }
+
+    before do
+      login_as(user)
+      visit checkout_path
+    end
+
+    describe "purchasing" do
+      context "takes us to the Payment Method" do
+        before do
+          split_fill_out_details
+          split_fill_out_billing_address
+        end
+
+        context "selecting a pick-up shipping method and submiting the form" do
+          before do
+            choose free_shipping.name
+          end
+
+          it "redirects the user to the Payment Method step" do
+            fill_in 'Any comments or special instructions?', with: "SpEcIaL NoTeS"
+            click_button "Next - Payment method"
+            expect(page).to have_current_path("/checkout/payment")
+          end
+        end
+
+        context "selecting a delivery shipping method and submiting the form" do
+          before do
+            choose shipping_with_fee.name
+            uncheck "ship_address_same_as_billing"
+          end
+
+          it "redirects the user to the Payment Method step" do
+            split_fill_out_shipping_address
+            fill_in 'Any comments or special instructions?', with: "SpEcIaL NoTeS"
+            click_button "Next - Payment method"
+            expect(page).to have_current_path("/checkout/payment")
+          end
+        end
+      end
+    end
+  end
+
+  context "when I have an out of stock product in my cart" do
+    before do
+      variant.on_demand = false
+      variant.on_hand = 0
+      variant.save!
```

Comment:
These three lines could probably just be one; `variant.update!(on_demand: false, on_hand: 0)`.